storage/objects/download_byte_range.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_download_byte_range]
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// downloadByteRange downloads a specific byte range of an object to a file.
+func downloadByteRange(w io.Writer, bucket, object string, startByte int64, length int64, destFileName string) error {
+	// bucket := "bucket-name"
+	// object := "object-name"
+	// startByte := 0
+	// length := 20
```

Comment:
I see in the Node sample you have start byte and end byte rather than length. Maybe it's good to do the same here and show how you would subtract to get length in the sample, for consistency? I just want it to be clear that the required params are different in Go.